### PR TITLE
Add instrumentation documentation for Spring Monitor

### PR DESCRIPTION
**Pull request by jeanbisutti**

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-starter-monitor/README.md [hunk 1]
```diff
@@ -83,9 +83,92 @@ You can then configure the connection string in two different ways:
 * With the `APPLICATIONINSIGHTS_CONNECTION_STRING` environment variable
 * With the `applicationinsights.connection.string` system property. You can use `-Dapplicationinsights.connection.string` or add the property to your `application.properties` file.
 
-### Additional instrumentations
-You can configure additional instrumentations with [OpenTelemetry instrumentations libraries](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks).
 
+### Configure the instrumentation
+
+The Spring starter will display the HTTP requests on the Application Insights resource in the Azure portal. You can configure additional instrumentation.
```
Comment (trask) [inline sdk/spring/spring-cloud-azure-starter-monitor/README.md:89]:
```suggestion
The Spring starter will capture HTTP requests by default. You can also configure additional instrumentation.
```

File: sdk/spring/spring-cloud-azure-starter-monitor/README.md [hunk 2]
```diff
@@ -83,9 +83,92 @@ You can then configure the connection string in two different ways:
...
+<dependencies>
+    <dependency>
+        <groupId>io.opentelemetry.instrumentation</groupId>
+        <artifactId>opentelemetry-jdbc</artifactId>
+        <version>{version}</version>
+    </dependency>
+</dependencies>
+```
+
+After, create a `DataSource` bean from `io.opentelemetry.instrumentation.jdbc.datasource.OpenTelemetryDataSource`:
```
Comment (trask) [inline sdk/spring/spring-cloud-azure-starter-monitor/README.md:105]:
```suggestion
Then wrap your `DataSource` bean in an `io.opentelemetry.instrumentation.jdbc.datasource.OpenTelemetryDataSource`, e.g.
```

File: sdk/spring/spring-cloud-azure-starter-monitor/README.md [hunk 3]
```diff
@@ -83,9 +83,92 @@ You can then configure the connection string in two different ways:
...
+    <dependency>
+        <groupId>io.opentelemetry.instrumentation</groupId>
+        <artifactId>opentelemetry-logback-appender-1.0</artifactId>
+        <version>{version}</version>
+        <scope>runtime</scope>
+    </dependency>
+</dependencies>
+```
+
+After, you have to enable the OpenTelemetry Logback appender. The following example shows you how to do this from a `logback.xml` file:
```
Comment (trask) [inline sdk/spring/spring-cloud-azure-starter-monitor/README.md:140]:
```suggestion
Then configure the OpenTelemetry Logback appender, e.g. in your `logback.xml` file:
```

File: sdk/spring/spring-cloud-azure-starter-monitor/README.md [hunk 4]
```diff
@@ -83,9 +83,92 @@ You can then configure the connection string in two different ways:
...
+        </encoder>
+    </appender>
+
+    <appender name="OpenTelemetry"
+              class="io.opentelemetry.instrumentation.logback.appender.v1_0.OpenTelemetryAppender">
+    </appender>
+
+    <root level="INFO">
+        <appender-ref ref="console"/>
+        <appender-ref ref="OpenTelemetry"/>
```
Comment (trask) [inline sdk/spring/spring-cloud-azure-starter-monitor/README.md:160]: have you run into any cyclical logging issues where the Azure Monitor exporter logs get captured and sent back to the Azure Monitor?

**Review by trask — APPROVED** [review]

File: sdk/spring/spring-cloud-azure-starter-monitor/README.md (hunk 4 above)
Comment (jeanbisutti) [reply sdk/spring/spring-cloud-azure-starter-monitor/README.md:160]: I have not noticed until now. I will verify when the self-diagnostics could enable the  Azure Monitor exporter logs at debug level.

**Pull request merged**